src/shared/watchdog.c

```diff
@@ -482,6 +482,23 @@ int watchdog_ping(void) {
         return watchdog_ping_now();
 }
 
+void watchdog_report_missing(void) {
+        /* If we failed to open the watchdog because the device doesn't exist, report why. If we successfully
+         * opened a device, or opening failed for a different reason, we logged then. But ENOENT failure may
+         * be transient, for example because modules being loaded or the hardware is being detected. This
+         * function can be called to log after things have settled down.
+         *
+         * If a device was specified explicitly, raise level. */
+
```

Comment:
i think if watchdog stuff is turned off explicitly we shouldn't log here

Reply:
We will get here only if we attempted to open the watchdog, which we should only do if it was configured. `watchdog_fd` starts out as `-EBADF`, and here we log iff `watchdog_fd == -ENOENT`.